multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ExtractBatchedServicesWithResolvedDynamicParametersStep.java

```diff
@@ -0,0 +1,130 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.common.util.MiscUtil;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudServiceInstanceExtended;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.ImmutableCloudServiceInstanceExtended;
+import org.cloudfoundry.multiapps.controller.core.cf.CloudHandlerFactory;
+import org.cloudfoundry.multiapps.controller.core.cf.v2.ServicesCloudModelBuilder;
+import org.cloudfoundry.multiapps.controller.core.helpers.DynamicResolvableParametersHelper;
+import org.cloudfoundry.multiapps.controller.core.model.DynamicResolvableParameter;
+import org.cloudfoundry.multiapps.controller.core.model.ImmutableDynamicResolvableParameter;
+import org.cloudfoundry.multiapps.controller.core.resolvers.v3.DynamicParametersResolver;
+import org.cloudfoundry.multiapps.controller.core.security.serialization.SecureSerialization;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.cloudfoundry.multiapps.mta.helpers.VisitableObject;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import com.sap.cloudfoundry.client.facade.CloudControllerClient;
+
+@Named("extractBatchedServicesWithResolvedDynamicParametersStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class ExtractBatchedServicesWithResolvedDynamicParametersStep extends SyncFlowableStep {
+
+    @Override
+    protected StepPhase executeStep(ProcessContext context) {
+        getStepLogger().debug(Messages.EXTACT_SERVICES_AND_RESOLVE_DYNAMIC_PARAMETERS_FROM_BATCH);
+
+        Set<DynamicResolvableParameter> dynamicResolvableParameters = context.getVariable(Variables.DYNAMIC_RESOLVABLE_PARAMETERS);
+        List<CloudServiceInstanceExtended> servicesCalculatedForDeployment = getServicesCalculatedForDeployment(context);
+
+        Set<DynamicResolvableParameter> dynamicParametersWithResolvedExistingInstances = resolveDynamicPramatersWithExistingInstances(context.getControllerClient(),
+                                                                                                                                      dynamicResolvableParameters,
+                                                                                                                                      servicesCalculatedForDeployment);
+
+        List<CloudServiceInstanceExtended> resolvedServiceInstancesWithDynamicParameters = servicesCalculatedForDeployment.stream()
```

Comment:
Maybe remove 'WithDynamicParameters' from this variable, at this point it's clear what you're resolving from the context :D